dist/stat.py

```diff
@@ -32,11 +34,44 @@ def check_unique_description(sorted_list):
             print("ERROR: repeated stat description in - '%s'" % (i.desc))
         temp = i.desc
 
+##########################################
+# Remove trailing digits for a string.
+##########################################
+def remove_suffix_digits(str):
+    return re.sub(r'\d+$', '', str)
+    
+##########################################
+# For each Stat subclass check the names are sorted in alphabetical order.
+##########################################
+def check_name_sorted(stat_list):
+    stat_dict = defaultdict(list)
+    for stat in stat_list:
+        stat_dict[type(stat)].append(stat)
+    for stat_type, stats in stat_dict.items():
+        # There're stats like hist_100 and hist_50 where alphabetically hist_100 comes first. 
+        # Thus we remove suffix digits and rely on the developer to order these stats correctly.
```

Comment:
I'd suggest the following comment to stay generic:
```suggestion
        # In alphabetical order, stat_name_100 comes before stat_name_50. For this reason, remove any numerical suffix before sorting the stats. Print an error if the stats are not sorted correctly.
```